org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalProvider.java

```diff
@@ -61,16 +61,18 @@ public class IdeContentProposalProvider {
 
 	/**
 	 * Create content assist proposals and pass them to the given acceptor.
+	 * @return true if the acceptor accepted all the proposals, false otherwise.
 	 */
-	public void createProposals(Collection<ContentAssistContext> contexts, IIdeContentProposalAcceptor acceptor) {
```

Comment:
This is not binary compatible. Do you strictly need the boolean return value or would it be possible to communicate this differently?

Reply:
I did not think about binary compatibility. 

One alternative I can think about would be to add a method `didAcceptedAllProposals` to `IIdeContentProposalAcceptor`. That would not break binary compatibility as far as I understand. If we also add a default implementation, it should not break source compatibility either, the problem three is that I would not know if we should return false or true.

So to keep both probably the cleanest would be to add a new interface `IIdeContentProposalAcceptor2` that inherit from `IIdeContentProposalAcceptor` and adds a `didAcceptedAllProposals`. Should I go for that one?